docs/05_Mock_tests.md

```diff
@@ -0,0 +1,57 @@
+# Mock Tests
+
+A web crawler is, by its very nature, dependent on the external service which it is crawling.
+In order to have a reliable CI pipeline, this external dependency needs to be removed.
+For this, there is a system in place to automatically save the requests made to a service, as well as the responses received.
+These can then be used in the CI pipeline to reliably test changes made to the Extractor, and avoid test failures due to API changes on the side of the service.
+
+## Multiple downloader implementations
+
+There are multiple implementations of the abstract class `Downloader`
+
+1. `DownloaderTestImpl` is used for running the test against the actual service.
+2. `RecordingDownloader` is used to save the request and response to a file, thus creating the mock.
+3. `MockDownloader` is used to answer requests using the saved mocks.
+
+### Usage
+
+There are 2 ways to specify which downloader should be used.
+
+First one is passing the `-Ddownloader=<value>` argument from the command line, where `value` can be one of
+[DownloaderType](https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/extractor/src/test/java/org/schabi/newpipe/downloader/DownloaderType.java)
+. The main use case is in the CI pipeline, like this: `./gradlew check --stacktrace -Ddownloader=MOCK`.  
+Other than that it can also be used to mass generate mocks by specifying which package should be tested. For example, if
+one wanted to update all YouTube mocks:
+`gradle clean test --tests 'org.schabi.newpipe.extractor.services.youtube.*' -Ddownloader=RECORDING`
+
+The second way is changing the field `DownloaderFactory.DEFAULT_DOWNLOADER`.
+The default value is `DownloaderType.REAL` which should be changed on the master branch.
```

Comment:
should or shouldn't? I don't see why downloader type shouldbe changed on the master branch

Reply:
Ehhh, `shoud not`

Reply:
```suggestion
The default value is `DownloaderType.REAL` which should not be changed on the master branch.
```